SrfGen/NHM/nhm2srf.py

```diff
@@ -37,301 +39,207 @@ def rand_shyp_dhyp(length = 1.0, width = 1.0):
-                strikes.append(bearing)
-                stk_norm += 1
-            else:
-                strikes.append((bearing + 180) % 360)
-                stk_rev += 1
-        if stk_norm and stk_rev:
-            print('WARNING: FAULT GOES BACK '
-                    'IN REVERSE OF ORIGINAL DIRECTION: %s' \
-                    % (name))
-        # assuming no reverse angles and ordering in one direction
-        if stk_rev:
-            mids = mids[::-1]
-            lengths = lengths[::-1]
-            strikes = strikes[::-1]
-        trace_length = sum(lengths)
-
-        # wanted parameters to override
-        t_hypo = 'n'
-        n_hypo = args.nhypo
-        n_slip = args.nslip
-        dhypos = args.dhypo
-        if faults != None:
-            fault = faults[fault_names.index(name)]
-            if len(fault) >= 2:
-                # given as hypocentre every x km
-                if fault[1][-1] == 'k':
-                    t_hypo = 'k'
-                    hyp_step = float(fault[1][:-1])
-                    n_hypo = 1 + int(trace_length // hyp_step)
-                    # 0th hypocentre position
-                    if n_hypo == 1:
-                        z_hypo = trace_length / 2.
-                    else:
-                        z_hypo = (trace_length % hyp_step) / 2.
-
-                # given as number of randomly placed hypocentres
-                elif fault[1][-1] == 'r':
-                    t_hypo = 'r'
-                    n_hypo = int(fault[1][:-1])
-
-                # given as number of hypocentres
-                else:
-                    n_hypo = int(fault[1])
-
-            if len(fault) >= 3:
-                n_slip = int(fault[2])
-            if len(fault) >= 4:
-                dhypos = list(map(float, fault[3].split(',')))
-        if t_hypo == 'n':
-            hyp_step = trace_length / (n_hypo * 2.)
-        seed = args.seed
-
-        # fixed values
-        dt = 0.025
-        cases = ['combined']
-        nseg = [n_plane]
-        seg_delay = [0]
-        mom = [-1]
-        rvfac_seg = ['-1']
-        gwid = ['-1']
-        rup_delay = [0]
-
-        flen = [lengths]
-        # subfault density
-        dlen = [[0.1] * n_plane]
-        # Karim: add 3km to depth if bottom >= 12km
-        if float(db[dbi + 6].split()[0]) >= 12:
-            raw_fwid = [[(float(db[dbi + 6].split()[0]) - dtop[0][0] + 3) \
-                / math.sin(math.radians(dip[0][0]))] * n_plane]
-        else:
-            raw_fwid = [[(float(db[dbi + 6].split()[0]) - dtop[0][0]) / math.sin(math.radians(dip[0][0]))] * n_plane]
-        fwid = []
-        for segment in raw_fwid:
-            fwid.append([round_subfault_size(f, mag) for f in segment])
-
-        if tect_type == "SUBDUCTION_INTERFACE":
-            mag = [skarlatoudis(fwid[0][0] * trace_length)]
+    # Read through the NHM file to find the details for our fault
+    while dbi < dbl and db[dbi] != fault_name:
```

Comment:
Better to read the NHM file once and save it to an dictionary of objects.

Reply:
Would this be slow for smaller FSFs?

Reply:
Pls

Reply:
lol - yeah. Fair. A lazy loading NHM reader function would be nice then though.